src/mscorlib/src/System/Array.cs

```diff
@@ -651,11 +655,22 @@ public void SetValue(Object value, params long[] indices)
         [ReliabilityContract(Consistency.WillNotCorruptState, Cer.Success)]
         public extern int GetUpperBound(int dimension);
 
+        private int _lowerBound0 = 0;
```

Comment:
Yep, it's a problem. Fields cannot be added to the `Array` class as its layout (or to be precise, the layout of types derived from `Array`) is determined by the runtime. And even if it was possible you certainly don't want to increase the size of array objects by adding a field.